diffkemp/semdiff/caching.py

```diff
@@ -440,21 +440,20 @@ def mark_uncachable_from_assumed_equal(self):
                     predecessor.cachable = False
                     vertex.prevents_caching_of.append(predecessor)
 
-    def _edge_callstack_to_string(self, callstack):
-        """Converts a callstack consisting of Edge objects to a string
-        representation."""
-        return "\n".join(["{} at {}:{}".format(call.target_name,
-                                               call.filename,
-                                               call.line)
-                          for call in callstack])
-
-    def _yaml_callstack_to_string(self, callstack):
+    def _edge_callstack_to_dict(self, callstack):
```

Comment:
Maybe this should be a static method if it stays here. Another option would be to create a `Callstack` class which would hold the dictionary and can be constructed from `Edge` objects.

Reply:
Thanks. I have created the class `Callstack` with a field `calls`  which is the list of calls. I have also changed the method `_edge_callstack_to_dict` to class method of the class. I have also created a method for getting string representation and YAML representation of the call stack which accepts path of snapshot for removing the snapshot prefix of file paths.